pandas/io/tests/test_parsers.py

```diff
@@ -2690,6 +2690,41 @@ def test_uneven_lines_with_usecols(self):
         df = self.read_csv(StringIO(csv), usecols=usecols)
         tm.assert_frame_equal(df, expected)
 
+    def test_read_empty_with_usecols(self):
+        # See gh-12493
+        names = ['Dummy', 'X', 'Dummy_2']
+        usecols = names[1:2]  # ['X']
+
+        def check_usecols(stringIO, expected):
+            df = self.read_csv(stringIO, names=names, usecols=usecols)
+            tm.assert_frame_equal(df, expected)
+
+        # first, check to see that the response of
+        # parser when faced with no provided columns
+        # throws the correct error, with or without usecols
+        errmsg = "No columns to parse from file"
+
+        with tm.assertRaisesRegexp(ValueError, errmsg):
```

Comment:
@gfyoung and that I am ok with. Or at least once we have a nice example we can decide whether its too big for 0.18.1 or not. But this needs to happen.

So all that said I know I said putting these together is a good idea (the usecols fix and error message changes). 

But might make sense to break these out into 2 PRs. (usecols changes on current master).

Then you can base your changes on that PR (for error message consistency).